flux-typeck/src/checker.rs

```diff
@@ -308,6 +319,31 @@ impl<'a, 'tcx, P: Phase> Checker<'a, 'tcx, P> {
         Ok(())
     }
 
+    fn is_stmt_nop(stmt: &Statement) -> bool {
```

Comment:
I'd define this as a method of `Statement` and call it `is_nop`. Also, you can follow clippy suggestion.

```rust
fn is_nop(&self) -> bool {
    matches!(self.kind, StatementKind::Nop)
}
```